### DISCUSSION TITLE
DgsQueryExecutor can not access HttpSession

### DISCUSSION THREAD

**Discussion by hantsy** [Netflix/dgs-framework#454] [General]
In my `signIn` data fetcher,  there are some codes accessing the web request.

```kotlin
val req = dfe.getDgsContext().requestData as DgsWebMvcRequestData
val session = req.webRequest?.sessionMutex as HttpSession
```

When writing test with `DgsQueryExecutor `.

```kotlin
  val loggedInQuery = "query signIn(\$input: Credentials!){signIn(credentials:\$input){ name roles  token }}"
  val token = dgsQueryExecutor.executeAndExtractJsonPath<String>(
      loggedInQuery,
      "data.signIn.token",
      mapOf("input" to mapOf("username" to "user", "password" to "password"))
  )
```
Got an NPE exception like this.

```bash
java.lang.NullPointerException: null cannot be cast to non-null type javax.servlet.http.HttpSession
```

The **signIn** query is working well in the **GraphiQL UI**.

**Comment by paulbakker** [comment]
Using `DgsQueryExecutor` directly (e..g. for tests) makes it not go through any web stack. You can pass a WebRequest and such manually through its arguments, but the framework won't do anything to initialize them. This means something like a "session" won't be there unless you manually set that up in the WebRequest instance you pass in.
This is by design, so that we can run queries without involving the web stack.

If you do want to go through the HTTP handlers, you should start a server and use the GraphQL client to send queries.
